Sources/SwiftDocC/Utility/SymbolExtensions.swift

```diff
@@ -0,0 +1,38 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2023 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import SymbolKit
+
+extension SymbolGraph.Symbol {
+
+    /// Return the offset of this symbol's doc comment in the original source code file,
+    /// adjusted for the interface language as follows:
+    ///
+    /// - Objective-C: subtract one from the start/end lines, and the start/end characters
+    /// - Swift and other languages: return the original offset
+    ///
+    /// - Returns: A ``SourceRange`` or nil if this symbol has no doc comment.
+    func offsetAdjustedForInterfaceLanguage() -> SymbolGraph.LineList.SourceRange? {
+        guard let range = docComment?.lines.first?.range else {
+            return nil
+        }
+        // Return the original range for Swift and other languages
+        guard identifier.interfaceLanguage == "objective-c" else {
```

Comment:
Should we also support source language aliases here?
```suggestion
        guard SourceLanguage(knownLanguageIdentifier: identifier.interfaceLanguage) == .objectiveC else {
```

Reply:
Actually, is there something general we could do here that's not conditional on the source language?

Reply:
I'll take the suggested change - thanks!